pkg/controllers/dynakube/proxy/reconciler_test.go

```diff
@@ -44,15 +62,38 @@ func newTestReconcilerWithInstance(client client.Client) *Reconciler {
 	return r
 }
 
+func newTestReconcilerWithCustomDynaKube(client client.Client, dk *dynakube.DynaKube) controllers.Reconciler {
+	r := NewReconciler(client, client, dk)
+
+	return r
+}
+
+func createDynaKubeWithProxy(proxy *dynakube.DynaKubeProxy) *dynakube.DynaKube {
+	return &dynakube.DynaKube{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNamespace,
+			Name:      testDynakubeName,
+		},
+		Spec: dynakube.DynaKubeSpec{
+			APIURL:     "https://testing.dev.dynatracelabs.com/api",
+			ActiveGate: dynakube.ActiveGateSpec{Capabilities: []dynakube.CapabilityDisplayName{dynakube.KubeMonCapability.DisplayName}},
+			Proxy:      proxy,
+		},
+	}
+}
+
 func TestReconcileWithoutProxy(t *testing.T) {
 	t.Run(`reconcile dynakube without proxy`, func(t *testing.T) {
-		r := newTestReconcilerWithInstance(fake.NewClientBuilder().Build())
+		testClient := fake.NewFakeClient()
+		r := newTestReconciler(testClient)
 		err := r.Reconcile(context.Background())
 
 		require.NoError(t, err)
 
 		var proxySecret corev1.Secret
-		err = r.client.Get(context.Background(), client.ObjectKey{Name: BuildSecretName(testDynakubeName), Namespace: testNamespace}, &proxySecret)
+
+		mockK8sClient := dtfake.NewClient()
+		err = mockK8sClient.Get(context.Background(), client.ObjectKey{Name: BuildSecretName(testDynakubeName), Namespace: testNamespace}, &proxySecret)
```

Comment:
here, you use a different client than what the `Reconciler` uses, so it will never have the secret, no matter what the `Reconciler` does

```suggestion
		err = testClient.Get(context.Background(), client.ObjectKey{Name: BuildSecretName(testDynakubeName), Namespace: testNamespace}, &proxySecret)
```

Reply:
done with `r.(*Reconciler).client` https://github.com/Dynatrace/dynatrace-operator/pull/3875/commits/1c4abfe681211d132ec48698079d2618271bdec9